incubator/virtualcluster/pkg/vn-agent/server/translate.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"strings"
+
+	"github.com/emicklei/go-restful"
+)
+
+// TranslatePath translate the naming between tenant and master cluster.
+func TranslatePath(req *restful.Request, tenantName string) string {
```

Comment:
PathFromTenantToMaster.

Let's make the func naming more descriptive.

Reply:
Also it is strange to take in an http request and translate it to a string type. the abstraction is not symmetric.

Reply:
good suggestion

Reply:
`restful.Request` here because it holds the route info it matched. I changed it to an object mutable method here.